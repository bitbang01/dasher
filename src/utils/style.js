function isNumber(value) {
  return typeof value === 'number' && !isNaN(value)
}

function em(value) {
  return isNumber(value) ? value + 'em' : value
}

function px(value) {
  return isNumber(value) ? value + 'px' : value
}

export function mq(breakpoint) {
  return `@media screen and (min-width: ${em(breakpoint)})`
}
