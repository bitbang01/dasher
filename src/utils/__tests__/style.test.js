import { mq } from '../style'

describe('mq()', () => {
  it('uses em units by default', () => {
    expect(mq(32)).toMatchSnapshot()
  })

  it('works with px units', () => {
    expect(mq('1200px')).toMatchSnapshot()
  })
})
