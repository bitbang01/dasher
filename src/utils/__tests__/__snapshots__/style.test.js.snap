// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mq() uses em units by default 1`] = `"@media screen and (min-width: 32em)"`;

exports[`mq() works with px units 1`] = `"@media screen and (min-width: 1200px)"`;
